ffmpeg/dag.py

```diff
@@ -69,31 +70,38 @@ def incoming_edge_map(self):
         raise NotImplementedError()
 
 
-DagEdge = namedtuple('DagEdge', ['downstream_node', 'downstream_label', 'upstream_node', 'upstream_label'])
+DagEdge = namedtuple('DagEdge', ['downstream_node', 'downstream_label', 'upstream_node', 'upstream_label', 'upstream_selector'])
 
 
 def get_incoming_edges(downstream_node, incoming_edge_map):
     edges = []
-    for downstream_label, (upstream_node, upstream_label) in list(incoming_edge_map.items()):
-        edges += [DagEdge(downstream_node, downstream_label, upstream_node, upstream_label)]
+    # downstream_label, (upstream_node, upstream_label) in [(i[0], i[1][:2]) for i in self.incoming_edge_map.items()]
+    for downstream_label, upstream_info in [(i[0], i[1]) for i in incoming_edge_map.items()]:
+        upstream_node, upstream_label = upstream_info[:2]
+        upstream_selector = None if len(upstream_info) < 3 else upstream_info[2]
```

Comment:
Where does the magical `3` constant come from? Needs explanation or named constant.

Reply:
or split it out to make it more clear:
```
if len(unsptream_info) < 3:  # because triangles have three sides (or whatever explanation)
    upstream_selector = None
else:
    upstream_selector = upstream_info[2]
```